Update from UPV

jsroldan opened the pull request:
Changelog:

- Fix computation of BLER
- Added new GPS logs (speed)
- Added PDSCH EVM.
- Resynchronization logic to be faster, tries to attach into the same frequency, no search.
- New CSV structure. Better format for postprocessing in numeric programs like Octave/Matlab
- Added auxiliar ringbuffer to improve the writing I/Q samples into HDD.
- Added new backend functions for rt-mbms-wui.

jsroldan committed:
Fix deleted thread acces, and change SDR start time

jsroldan committed:
Updated srsRAN submodule commit

jsroldan committed:
Added fractional intervals less than a second of time measurement

jsroldan committed:
GPS speed logging added

jsroldan committed:
Magic numbers eliminated, convert it into variables

jsroldan committed:
Added PDSCH EVM measurement, deleted BER

jsroldan committed:
Changed conf values, modiefied teh average subframes period for CNR

jsroldan committed:
Resynchronization change to be faster, tries to attach to same Cell ID

The recevier doesn't search every cell ID when loses the synchrononization, tries to connect to the same already known cell.

jsroldan committed:
Fix BLER calculation, new CSV structure. New argument to choose to repeat the reading of a sample file.

The computation of the BLER was erronious, it has been fixed. Also the columns and structure on how the data is saved in the CSV has been change, now every time there's a new measurment a new row is saved in the file indicating the status of the receiver (SYNCING, SEARCHING, o printing the data if it's processing).

jsroldan committed:
Added the hekper functions to get and set the different values of phy and processors within the rt-wui

jsroldan committed:
Added an auxiliar ringbuffer to fix the overflow problem when saving the samples to a disk.
With this secondary buffer, the chunks to be saved are bigger, and this way minimizes the blocking of writting when saving by small chunks.

jsroldan committed:
Fix a error in rest_handler intialization in main

jsroldan committed:
Fix some errors with the RestHandler, added almost all new functionlity for the rt-wui
Only pendind things are the displayed of cir and correlation.

dsilhavy commented:
@alessandro-lucco: Fyi, this PR includes the changes presented by @jsroldan in the last call. If you find the time, it would be nice if you can run some tests on your side as well. We are currently testing against sample recordings: https://5g-mag.github.io/Getting-Started/pages/lte-based-5g-broadcast/additional/sample-files.html

alessandro-lucco commented:
> @alessandro-lucco: Fyi, this PR includes the changes presented by @jsroldan in the last call. If you find the time, it would be nice if you can run some tests on your side as well. We are currently testing against sample recordings: https://5g-mag.github.io/Getting-Started/pages/lte-based-5g-broadcast/additional/sample-files.html

Thanks Daniel and Jaime, I did some tests in the last day. 
With different sample files it works very well. Using an SDR card with live signal it receives correctly but after a few seconds it returns a segmentation fault. I am in contact with Jaime to try to understand the problem. I do not know if others have had the opportunity to test it. 
The setup I used is a PC with Ubuntu 22, 16GB of RAM and an i5 processor, as SDR I tested both with BladeRF and with USRP B210. I tested also on Ubuntu 20 and and I see the same behavior.

dsilhavy commented:
> > @alessandro-lucco: Fyi, this PR includes the changes presented by @jsroldan in the last call. If you find the time, it would be nice if you can run some tests on your side as well. We are currently testing against sample recordings: https://5g-mag.github.io/Getting-Started/pages/lte-based-5g-broadcast/additional/sample-files.html
> 
> Thanks Daniel and Jaime, I did some tests in the last day. With different sample files it works very well. Using an SDR card with live signal it receives correctly but after a few seconds it returns a segmentation fault. I am in contact with Jaime to try to understand the problem. I do not know if others have had the opportunity to test it. The setup I used is a PC with Ubuntu 22, 16GB of RAM and an i5 processor, as SDR I tested both with BladeRF and with USRP B210. I tested also on Ubuntu 20 and and I see the same behavior.

Thanks for testing @alessandro-lucco . I only tested a sample file so far and that worked fine. Where does the live signal come from? Is it the 5G-MAG transmitter, or are you using a commercial solution like R&S? Do you see the same problem with the version that does not include the changes by @jsroldan ?

alessandro-lucco commented:
> > > @alessandro-lucco: Fyi, this PR includes the changes presented by @jsroldan in the last call. If you find the time, it would be nice if you can run some tests on your side as well. We are currently testing against sample recordings: https://5g-mag.github.io/Getting-Started/pages/lte-based-5g-broadcast/additional/sample-files.html
> > 
> > 
> > Thanks Daniel and Jaime, I did some tests in the last day. With different sample files it works very well. Using an SDR card with live signal it receives correctly but after a few seconds it returns a segmentation fault. I am in contact with Jaime to try to understand the problem. I do not know if others have had the opportunity to test it. The setup I used is a PC with Ubuntu 22, 16GB of RAM and an i5 processor, as SDR I tested both with BladeRF and with USRP B210. I tested also on Ubuntu 20 and and I see the same behavior.
> 
> Thanks for testing @alessandro-lucco . I only tested a sample file so far and that worked fine. Where does the live signal come from? Is it the 5G-MAG transmitter, or are you using a commercial solution like R&S? Do you see the same problem with the version that does not include the changes by @jsroldan ?

At the moment i can only test with R&S transmitter. No, with the other versions I have always done the same steps (with same input signal) and I did not encounter segmentation faults, except in very few cases. I think it's a buffer issue related to the SDR management.

jsroldan commented:
The next week I'll debug it to try to figure out what is wrong, I've never tested with a SDR so thanks @alessandro-lucco for the finding!

dsilhavy commented:
Comment from the call by @alessandro-lucco : No 5GBC signal needed, but SDR needs to be connected to reproduce the issue that @alessandro-lucco reported above.

jsroldan commented:
Hi all, I finally found the error, the modem stored the samples in the writing buffer although writing the I/Q samples to a file was disable, so basically the buffer was been fulfilled but never emptied (read), no checks if writing is enable to do that.

jsroldan committed:
Fixed an error related to the storing of the I/Q samples to a file.

The samples were saved to the writing buffer even when the writing was disable, causing an overflow.

jordijoangimenez commented:
We wait for confirmation for merging this PR after telco between @jsroldan and @alessandro-lucco

jsroldan commented:
Last week, Alexandro and I had a meeting to check the issues Alexandro found. We can confirm that are a few things that I need to have a look at. The sample rate is not choose correctly (which happens also in previous versions), and the writing of the samples to a file present some problems.

jsroldan committed:
Add missing option to allow non contiguous RRC SN across multiple MCH scheduling periods

Missing from previous version made by Klaus.

jsroldan commented:
Hi @alessandro-lucco and @dsilhavy, I've added a missing option that somehow I forgot to add when I started updating the modem.

I've tested it with our rt-mbms-tx and it works perfectly, the content is received and display without any cuts or missing packets.

@alessandro-lucco I encourage you to test it with your R&S live signal, it should work. If you use it with R&S tx, the option "allow_rrc_sn_across_periods" has to be set to false, and with the 5G MAG rt-mbms-tx it need to be set to true. 

Any questions don't hesitate to ask.

alessandro-lucco commented:
> Hi @alessandro-lucco and @dsilhavy, I've added a missing option that somehow I forgot to add when I started updating the modem.
> 
> I've tested it with our rt-mbms-tx and it works perfectly, the content is received and display without any cuts or missing packets.
> 
> @alessandro-lucco I encourage you to test it with your R&S live signal, it should work. If you use it with R&S tx, the option "allow_rrc_sn_across_periods" has to be set to false, and with the 5G MAG rt-mbms-tx it need to be set to true.
> 
> Any questions don't hesitate to ask.

Hi, I confirm that this version by @jsroldan works correctly in Release 14 with the signal coming from the R&S TX. 

Just a comment that is not related to this version, using a recent version of Cmake gives me an error on the minimum version reported in the [CMakeLists.txt](https://github.com/5G-MAG/rt-mbms-modem/blob/development/CMakeLists.txt) file and in the related files inside srsran. If the error is common I think the first row  (`cmake_minimum_required (VERSION 3.16)` ) of each files should be updated.

dsilhavy commented:
@alessandro-lucco 
Please provide information about your setup: Ubuntu version, CMake version and everything that can be useful

alessandro-lucco commented:
> @alessandro-lucco Please provide information about your setup: Ubuntu version, CMake version and everything that can be useful
Here some informations:

cmake version 3.31.7
Ubuntu 22.04.5 LTS


Warning obtained after cmake
```
CMake Error at lib/srsran/CMakeLists.txt:33 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.

```

jordijoangimenez committed:
Update CMakeLists.txt
